store/src/java/com/zimbra/cs/account/ldap/LdapProvisioning.java

```diff
@@ -10721,4 +10721,26 @@ public boolean isCacheEnabled() {
         return useCache;
     }
 
+    @Override
+    public void refreshUserCredentials(Account account) throws ServiceException {
+        ZLdapContext zlc = null;
+
+        try {
+            zlc = LdapClient.getContext(LdapServerType.MASTER, LdapUsage.GET_ENTRY);
+            String[] returnAttrs = {"userPassword", "zimbraAuthTokens", "zimbraAuthTokenValidityValue"};
+            ZAttributes attrs = helper.getAttributes(zlc, ((LdapEntry) account).getDN(), returnAttrs);
+            Map<String, Object> finalAttrs = account.getAttrs(false, false);
+            finalAttrs.putAll(attrs.getAttrs());
+            account.setAttrs(finalAttrs);
+            extendLifeInCacheOrFlush(account);  // Put this back into the cache
+        } catch (LdapException e) {
```

Comment:
Can combine these 2 catch blocks - `catch (LdapException | AccountServiceException e)`

Reply:
roger that.